File: examples/Gaussian_example.R
library(renv)
renv::activate()
library(regMMD)

#simulate data
x = rnorm(50,0,1.5)
# estimate the mean and variance (assuming the data is Gaussian)
Est = mmd_est(x, model="Gaussian")
# print a summary
summary(Est)
# estimate the mean (assuming the data is Gaussian with known standard deviation =1.5)
Est2 = mmd_est(x, model="Gaussian.loc", par2=1.5)
# print a summary
summary(Est2)
# estimate the standard deviation (assuming the data is Gaussian with known mean = 0)
Est3 = mmd_est(x, model="Gaussian.scale", par1=0)
# print a summary
summary(Est3)
# test of the robustness
x[42] = 100
mean(x)
# estimate the mean and variance (assuming the data is Gaussian)
Est4 = mmd_est(x, model="Gaussian")
summary(Est4)

File: examples/Gaussian_example.py
import numpy as np
from regmmd import MMDEstimator
from regmmd.models import GaussianLoc, GaussianScale, Gaussian

import cProfile
import pstats


def main():
    print("Estimating only the mean")
    rng = np.random.default_rng(seed=123)
    print("Sampling points..")
    x = rng.normal(loc=0, scale=1.5, size=50)
    print("Initializing model")
    model = GaussianLoc(
        par2 = 1.5
    )
    print("Initializing estimator")
    mmd_estim = MMDEstimator(
        model=model,
        par1=None, 
        par2=1.5,
        kernel="Gaussian",
        solver={
            "type": "GD",
            "burnin": 500,
            "n_step": 1000,
            "stepsize": 1,
            "epsilon": 1e-4
        }
    )
    print("fitting estimator")
    res = mmd_estim.fit(
        X=x
    )
    print(res)

    print("Estimating both the mean and variance")
    # 
    print("Sampling points..")
    x = rng.normal(loc=0, scale=1.5, size=50)
    print("Initializing model")
    model = Gaussian(par1=0.2, par2=1.2, random_state=10)
    print("Initializing estimator")
    mmd_estim = MMDEstimator(
        model=model,
        par1=1.0, 
        par2=0.2,
        kernel="Gaussian",
        solver={
            "type": "SGD",
            "burnin": 500,
            "n_step": 1000,
            "stepsize": 1,
            "epsilon": 1e-4
        }
    )
    print("fitting estimator")
    res = mmd_estim.fit(
        X=x
    )
    print(res)

if __name__ == "__main__":
    main()
    # cProfile.run("main()", "profile_stats")
    # stats = pstats.Stats("profile_stats")
    # stats.sort_stats(pstats.SortKey.TIME).print_stats(10)  # Show top 10 time-consuming functions